### PR TITLE
Implement `macaw-loader-riscv`

### PR DESCRIPTION
**Pull request by RyanGlScott**
Fixes https://github.com/GaloisInc/macaw-loader/issues/13.

### CODE REVIEW

**Comment by Ptival** [comment]
Looks like we favor HTTPS over SSH gitmodules in other repos, so I'm okay with this change.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -26,6 +26,12 @@ jobs:
       with:
         ghc-version: ${{ matrix.ghc-ver }}
         cabal-version: ${{ matrix.cabal }}
+    - name: System Dependencies
+      run: |
+        # Softfloat repo needs to be recursively cloned
+        cd submodules/softfloat-hs
+        git submodule update --init --recursive
```
Comment (kquick) [inline .github/workflows/ci.yaml:33]: Can we replace the `cd` activities with just `git -C submodules/softfloat-hs submodule ...`?

**Review by kquick — APPROVED** [review]
Just a small recommendation for the CI to avoid `cd` calesthenics.

Also as discussed offline, switching submodules to `https:` could result in hitting unauthenticated rate limiting, especially if we use local runners.  Not a current issue, but leaving this comment here in the hopes it will be helpful if we do have issues around this in the future.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (RyanGlScott) [reply .github/workflows/ci.yaml:33]: Oh yes, that's much better. I copied this part over from [`macaw`'s CI](https://github.com/GaloisInc/macaw/blob/1a2b9284f11eaa67420d29641afe3d32ae1418af/.github/workflows/ci.yaml#L57-L60), but I like your version more than what `macaw` is doing.

**Pull request merged**